python/tvm/relay/backend/compile_engine.py

```diff
@@ -63,6 +83,316 @@ def _get_cache_key(source_func, target):
     return source_func
 
 
+def get_shape(shape):
+    """Convert the shape to correct dtype and vars."""
+    ret = []
+    for dim in shape:
+        if isinstance(dim, tvm.expr.IntImm):
+            val = int(dim)
+            assert val <= np.iinfo(np.int32).max
+            ret.append(tvm.expr.IntImm("int32", val))
+        elif isinstance(dim, tvm.expr.Any):
+            ret.append(tvm.var("any_dim", "int32"))
```

Comment:
why is this ok? suppose there is two Any in the code, do they get mapped to the same symbolic shape?

Reply:
Each time encounters Any, it will create a new var. These vars are different because they have different pointer.

Reply:
ooh, I forgot it is by pointer...